### PR TITLE
Refactor site filters

### PR DESCRIPTION
**Pull request by mikestone14**
This commit includes some potentially valuable changes:

* Sites are treated as arrays of strings everywhere except where they are needed as dropdown options. When we need them as dropdown options we use the pre-existing model method to convert them.
* Allows the user to refresh the page and maintain their filters.
* Fixes a bug that now allows all permitted sites to be displayed as dropdown options.

### CODE REVIEW

File: views/pages/ViewChartPage.jsx [hunk 1]
```diff
@@ -47,11 +60,16 @@ const ViewChartPage = () => {
   }
 
   useEffect(() => {
-    fetchGraph(chart_id).then((graph) => {
-      setGraph(graph)
-      reset(graph.filters)
+    const parsedQuery = qs.parse(search.replace(/^\?/, ''))
```
Comment (mikestone14) [inline views/pages/ViewChartPage.jsx:63]: the search string has a `?` prepended to it, which borks `qs.parse`.

File: server/controllers/chartsDataController/index.js [hunk 2]
```diff
@@ -6,18 +6,17 @@ import SubjectModel from '../../models/SubjectModel'
...
+    const userSites = StudiesModel.sanitizeAndSort(req.session.userAccess)
     const { chart_id } = req.params
-    const { userAccess } = req.session
     const parsedQueryParams = qs.parse(req.query)
-    const filters = FiltersModel.calculateFilters(
-      parsedQueryParams.filters,
-      userAccess
-    )
+    const filtersService = new FiltersService(parsedQueryParams.filters)
+    const filters = filtersService.sanitize(userSites)
```
Comment (Elkrival) [inline server/controllers/chartsDataController/index.js]: Do we want to instantiate the services here and then again in the SubjectModel.allForAssessment..?

File: views/pages/ViewChartPage.jsx [hunk 3]
```diff
@@ -47,11 +60,16 @@ const ViewChartPage = () => {
...
+
+    fetchGraph(chart_id, parsedQuery.filters).then((newGraph) => {
+      setGraph(newGraph)
+      reset({
+        ...newGraph.filters,
+        sites: StudiesModel.dropdownSelectOptions(newGraph.filters.sites),
```
Comment (Elkrival) [inline views/pages/ViewChartPage.jsx:69]: This logic is being used twice, I wonder if we can assign it to a variable..?

**Review by Elkrival — COMMENTED** [review]
a few questions..

File: views/pages/ViewChartPage.jsx (hunk 3 above)
Comment (mikestone14) [reply views/pages/ViewChartPage.jsx:69]: what would that look like? I see that we're using it twice but it seems like we need to (unless we don't need to call `reset`)? Do you know why we need to reset the form?

File: server/controllers/chartsDataController/index.js (hunk 2 above)
Comment (mikestone14) [reply server/controllers/chartsDataController/index.js]: hmm yeah I'm not sure. I could go either way. What do you think about this if we want to only call it once? https://github.com/AMP-SCZ/dpdash/pull/442/commits/e43b38f0a442365c5be74f263852c5867477f79a

Comment (Elkrival) [reply server/controllers/chartsDataController/index.js]: That's what I was struggling with for a while. Is passing a service in that manner a good pattern

File: views/pages/ViewChartPage.jsx (hunk 3 above)
Comment (Elkrival) [reply views/pages/ViewChartPage.jsx:69]: I think with what we did with the service we can do it all from the useEffect

Comment (Elkrival) [reply views/pages/ViewChartPage.jsx:69]: reset is used when we need to hydrate form values from an async call

**Review by Elkrival — APPROVED** [review]
LGTM

File: views/pages/ViewChartPage.jsx (hunk 3 above)
Comment (mikestone14) [reply views/pages/ViewChartPage.jsx:69]: that service is on the server side, so we know the data is accurate we just need to convert it to dropdown options.

**Pull request merged**